@here/harp-mapview/lib/Tile.ts

```diff
@@ -989,27 +988,19 @@ export class Tile implements CachedResource {
     private updateBoundingBox(newBoundingBox?: OrientedBox3) {
         if (newBoundingBox) {
             this.m_boundingBox.copy(newBoundingBox);
-
-            // Update geo box to match the given bounding box.
-            const tmpPos = this.m_boundingBox.position.clone().add(this.m_boundingBox.extents);
-            this.geoBox.northEast.copy(this.mapView.projection.unprojectPoint(tmpPos));
-            tmpPos.copy(this.m_boundingBox.position).sub(this.m_boundingBox.extents);
-            this.geoBox.southWest.copy(this.mapView.projection.unprojectPoint(tmpPos));
+            this.m_worldCenter.copy(this.boundingBox.position);
         } else {
             this.projection.projectBox(this.geoBox, this.boundingBox);
```

Comment:
I think that this.m_worldPos should be also updated here, after this.geoBox projection. Let's consider scenario:
1. Tile has been decoded.
2. Bounding box not provided via decoded tile.
3. GeoBox has been updated (elevated) based on decodedTile.maxGeometryHeight
4.  Method updateBoundingBox() gets called with undefined parameter, so this.boundingBox gets updated, but world position not.

Please check, is this intended?

Reply:
The tile center used to compute local tile coordinates in the decoder is not elevated. If we update the tile center so that it matches the center of the elevated bounding box, the geometry will be rendered above of where it should actually be.

When a boudingbox is provided by the decoder though, we assume that the geometry is in local coordinates using the bounding box center as reference. So there we do update the tile center.

Reply:
Ok, understood the reasons now... so that was probably the cause of bug. That's all fine, it is little pity that you did not added such explanation (maybe in updateBoundingBox) so everything is clear for the other devs.